packages/backend/src/workers/provider/LlamaCppPython.ts

```diff
@@ -197,9 +197,21 @@ export class LlamaCppPython extends InferenceProvider {
     if (this.configurationRegistry.getExtensionConfiguration().experimentalGPU) {
       const gpus: IGPUInfo[] = await this.gpuManager.collectGPUs();
       if (gpus.length === 0) throw new Error('no gpu was found.');
-      if (gpus.length > 1)
-        console.warn(`found ${gpus.length} gpus: using multiple GPUs is not supported. Using ${gpus[0].model}.`);
-      gpu = gpus[0];
+      let selectedGPU = 0;
+      if (gpus.length > 1) {
+        // Look for a GPU that is of a known type, use the first one found.
+        // Fall back to the first one if no GPUs are of known type.
+        for (let i = 0; i < gpus.length; i++) {
+          if (gpus[i].vendor !== GPUVendor.UNKNOWN) {
+            selectedGPU = i;
+            break;
+          }
+        }
```

Comment:
nit: use the Array.find method ?

Reply:
I think we would need to use findIndex versus find. With that we'd have to check for -1 as that is what's returned if there is no match. Not sure it would be much less code or clearer.

Reply:
If there is a single GPU we will take it but won't check if the vendor is known. Should we check it as well ?

Reply:
@jeffmaury I left it not checking since that was the way it was before and I was not sure if that would break anything. I can add the check if that is the right thing to do.

Reply:
Would be good because I understood your configuration have 2 GPUs returned one which is not to be selected so I wonder we could have a configuration where we could have this single GPU returned